scripts/idl/generators/java/__init__.py

```diff
@@ -76,19 +82,231 @@ def CallbackName(attr: Attribute, cluster: Cluster, known_enum_types: List[matte
+
+    def without_list(self):
+        result = self.clone()
+        result.attrs.remove(EncodableValueAttr.LIST)
+        return result
+
+    def get_underlying_struct(self):
+        s = self.context.find_struct(self.data_type.name)
+        if not s:
+            raise Exception("Struct %s not found" % self.data_type.name)
+        return s
+
+    def get_underlying_enum(self):
+        e = self.context.find_enum(self.data_type.name)
+        if not e:
+            raise Exception("Enum %s not found" % self.data_type.name)
+        return e
+
+    @property
+    def boxed_java_type(self):
+        t = ParseDataType(self.data_type, self.context)
+
+        if type(t) == FundamentalType:
+            if t == FundamentalType.BOOL:
+                return "Boolean"
+            elif t == FundamentalType.FLOAT:
+                return "Float"
+            elif t == FundamentalType.DOUBLE:
+                return "Double"
+            else:
+                raise Error("Unknown fundamental type")
+        elif type(t) == BasicInteger:
+            if t.byte_count >= 4:
+                return "Long"
+            else:
+                return "Integer"
+        elif type(t) == BasicString:
+            if t.is_binary:
+                return "byte[]"
+            else:
+                return "String"
+        elif type(t) == IdlEnumType:
+            return "Integer"
+        elif type(t) == IdlBitmapType:
+            return "Integer"
+        else:
+            return "Object"
+
+    @property
+    def boxed_java_signature(self):
+        # Optional takes precedence over list - Optional<ArrayList> compiles down to just java.util.Optional.
+        if self.is_optional:
+            return "Ljava/util/Optional;"
+
+        if self.is_list:
+            return "Ljava/util/ArrayList;"
+
+        t = ParseDataType(self.data_type, self.context)
+
+        if type(t) == FundamentalType:
+            if t == FundamentalType.BOOL:
+                return "Ljava/lang/Boolean;"
+            elif t == FundamentalType.FLOAT:
+                return "Ljava/lang/Float;"
+            elif t == FundamentalType.DOUBLE:
+                return "Ljava/lang/Double;"
+            else:
+                raise Error("Unknown fundamental type")
+        elif type(t) == BasicInteger:
+            if t.byte_count >= 4:
+                return "Ljava/lang/Long;"
+            else:
+                return "Ljava/lang/Integer;"
+        elif type(t) == BasicString:
+            if t.is_binary:
+                return "[B"
+            else:
+                return "Ljava/lang/String;"
+        elif type(t) == IdlEnumType:
+            return "Ljava/lang/Integer;"
+        elif type(t) == IdlBitmapType:
+            return "Ljava/lang/Integer;"
+        else:
+            return "Lchip/devicecontroller/ChipStructs${}Cluster{};".format(self.context.cluster.name, self.data_type.name)
+
+
+def EncodableValueFrom(field: Field, context: TypeLookupContext) -> EncodableValue:
```

Comment:
Can we add more comments to these functions please? Otherwise, it's quickly going to become un-readable by other developers looking to help on this effort.

Reply:
Will do in a follow-up.